tensorflow/lite/micro/simple_memory_allocator.cc

```diff
@@ -113,10 +114,33 @@ uint8_t* SimpleMemoryAllocator::AllocateTemp(size_t size, size_t alignment) {
     return nullptr;
   }
   temp_ = aligned_result + size;
+  temp_buffer_ptr_check_sum_ =
+      temp_buffer_ptr_check_sum_ ^ (reinterpret_cast<intptr_t>(aligned_result));
```

Comment:
Is there a reason that this is not written as "temp_buffer_ptr_check_sum_ ^= reinterpret_cast<intptr_t>(aligned_result);", which should be equivalent and shorter? Same applies to lines 124~125 in SimpleMemoryAllocator::DeallocateTemp(uint8_t* temp_buf) below.

Reply:
Done. No strong reason other than my tendency of over verbosity sometimes.